app.py

```diff
@@ -524,37 +524,36 @@ def classpage(class_ok_id):
 
                     if not lecture.get('is_playlist'):
                         try:
+                            id = Class.add_lecture(cls, lecture, db)
                             transcript, preds = transcribe_utils.transcribe(
                                 mode=app.config['TRANSCRIPTION_MODE'],
                                 youtube_link=request.form['link'],
                                 youtube=youtube,
                                 transcription_classifier=ts_classifier,
                             )
-                            id = Class.add_lecture(cls, lecture, db)
                             Lecture.add_transcript(id, transcript, preds, db)
                         except ValueError as e:
                             flash('There was a problem retrieving the caption track for this video. {0}'.format(consts.NO_CAPTION_TRACK_MESSAGE))
```

Comment:
Looks fine to me. If you want to add "the lecture was added but there was a problem getting the caption track..." that may clear up some confusion.